src/TestSuite/Migrator.php

```diff
@@ -0,0 +1,185 @@
+            $migrations = new Migrations($config);
+            if ($this->isStatusChanged($migrations)) {
+                if (!in_array($connectionName, $connectionsToDrop)) {
+                    $connectionsToDrop[] = $connectionName;
+                }
+            }
+        }
+
+        if (empty($connectionsToDrop)) {
+            $this->io->success("No migration changes detected.");
+
+            return $this;
+        }
+
+        $schemaCleaner = new SchemaCleaner($this->io);
+        foreach ($connectionsToDrop as $connectionName) {
+            $schemaCleaner->dropTables($connectionName);
+        }
+
+        foreach ($configs as $migration) {
+            $this->runMigrations($migration);
+        }
+
+        // Truncate all created tables, except migration tables
+        foreach ($connectionsToDrop as $connectionName) {
+            $schema = ConnectionManager::get($connectionName)->getSchemaCollection();
+            $allTables = $schema->listTables();
+            $tablesToTruncate = $this->unsetMigrationTables($allTables);
+            $schemaCleaner->truncateTables($connectionName, $tablesToTruncate);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Unset the phinx migration tables from an array of tables.
+     *
+     * @param  string[] $tables
+     * @return array
+     */
+    protected function unsetMigrationTables(array $tables): array
+    {
+        $endsWithPhinxlog = function (string $string) {
+            $needle = 'phinxlog';
+            return substr($string, -strlen($needle)) === $needle;
+        };
+
+        foreach ($tables as $i => $table) {
+            if ($endsWithPhinxlog($table)) {
+                unset($tables[$i]);
+            }
+        }
+
+        return array_values($tables);
+    }
+
+    /**
+     * Checks if any migrations are up but missing.
+     *
+     * @param  Migrations $migrations
+     * @return bool
+     */
+    protected function isStatusChanged(Migrations $migrations): bool
+    {
+        foreach ($migrations->status() as $migration) {
+            if ($migration['status'] === 'up' && ($migration['missing'] ?? false)) {
+                $this->io->info('Missing migration(s) detected.');
+                return true;
+            }
+            if ($migration['status'] === 'down') {
+                $this->io->info('New migration(s) found.');
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Stringify the migration parameters.
+     * This is used to display readable messages
+     * on the command line.
+     *
+     * @param string[] $config Config array
+     * @return string
+     */
+    protected function stringifyConfig(array $config): string
+    {
+        $options = [];
+        foreach (['connection', 'plugin', 'source', 'target'] as $option) {
+            if (isset($config[$option])) {
+                $options[] = $option . ' "'.$config[$option].'"';
```

Comment:
```suggestion
                $options[] = $option . ' "' . $config[$option] . '"';
```

Our phpcs rules will get grumpy about the operator spacing.